plugins/net_plugin/net_plugin.cpp

```diff
@@ -912,6 +921,13 @@ namespace eosio {
 
       std::chrono::nanoseconds         connection_start_time{0};
 
+      // block nack support
+      static constexpr uint16_t consecutive_block_nacks_threshold{2}; // stop sending blocks when reached
```

Comment:
I think this could be as low as `0`. 0 worked fine in all my tests and I don't think a large value is useful here.

Also this PR, only doesn't block nack one connection. If we wanted to keep 2 or more peers sending us blocks the PR could be updated with that functionality. I don't personally think it is needed. And I think best to keep this as simple as possible. If we find this feature doesn't work it can be disabled via `p2p-disable-block-nack` and we can address any issues we find.